communication/src/TlsConnectionManager.cpp

```diff
@@ -341,6 +355,17 @@ void ConnectionManager::connect() {
   }
 }
 
+void ConnectionManager::handleConnectTimeout(NodeNum i, const std::error_code& error_code) {
+  asio::steady_timer& deadline_timer_ = connecting_.at(i).second;
+  if (deadline_timer_.expiry() <= asio::steady_timer::clock_type::now()) {
+    LOG_WARN(logger_, "Timeout during connecting to node : " << i);
+    connecting_.at(i).first.close();
+    status_->num_connecting = connecting_.size();
+  }
+  LOG_TRACE(logger_, "Deadline Timer end");
+  connecting_.erase(i);
```

Comment:
Why connecting_.erase(i) is called also in case the timeout is not expired (sporadic call)?
Don't you need to call async_wait again in this case?

Reply:
when timeout is not expired it will be called until we call in timer.cancel()

Reply:
According to the documentation, it is possible that the callback is called even when timeout is not expired yet, that's why we have 'if' in this function. You don't distinguish between this case and timer cancelation.

Reply:
You did not answer my second question (regarding async_wait).